source/fundamentals/crud/read-operations/text.txt

```diff
@@ -0,0 +1,229 @@
+movie franchise: 
+
+.. code-block:: json
+
+   { "_id": 1, "title": "2 Fast 2 Furious ", "tags": ["undercover", "drug dealer"] } 
+   { "_id": 2, "title": "Fast 5", "tags": ["bank robbery", "full team"] }
+   { "_id": 3, "title": "Furious 7", "tags": ["emotional"] }
+   { "_id": 4, "title": "The Fate of the Furious", "tags": ["betrayal"] }
+
+Text Index
+~~~~~~~~~~
+
+You must create a **text index** before running a text search. A text
+index specifies the string or string array field on which to run a text
+search.
+
+In the following examples, you run text searches on the ``title``
+field in the ``fast_and_furious_movies`` collection. To enable text
+searches on the ``title`` field, create a text index using the
+:ref:`Indexes <index-text-indexes>` builder with the following 
+snippet: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/SearchText.java
+   :language: java
+   :dedent:
+   :start-after: begin textIndex
+   :end-before: end textIndex
+
+For more information, see the :ref:`Text Indexes <textIndexes>` section
+of our Indexes guide. 
+
+Text Search
+-----------
+
+Use the ``Filters.text()`` method to specify a text search. 
+
+The ``Filters.text()`` method uses the :doc:`Filters builder
+</fundamentals/builders/filters>` to define a query filter specifying
+what to search for during the text search. The query filter is
+represented by a :ref:`Bson <bson>` instance. Pass the query filter to the
+``find()`` method to  run a text search. 
+
+When you execute the ``find()`` method, MongoDB runs a text search on
+all the fields indexed with the text index on the collection. MongoDB
+returns documents that contain one or more of the search terms and a
+:ref:`relevance score for each result <sorts-crud-text-search>`. 
+
+Specify Options
+~~~~~~~~~~~~~~~
+
+You can include ``TextSearchOptions`` as the second parameter of the
+``Filters.text()`` method to specify text search options such as case
+sensitivity. By default, text searches run without case sensitivity
+which means the search matches lowercase and uppercase values.
+
+To specify a case sensitive search, use the following snippet:
+
+.. code-block:: java
+
+   TextSearchOptions options = new TextSearchOptions().caseSensitive(true);
+   Bson filter = Filters.text("SomeText", options);
+
+For more information on the method and class in this section, see the following API documentation:
+
+- :java-core-api:`Filters.text() <com/mongodb/client/model/Filters.html#text(java.lang.String)>`
+- :java-core-api:`TextSearchOptions <com/mongodb/client/model/TextSearchOptions.html>`
+
+.. _term_search:
+
+Search Text by a Term
+~~~~~~~~~~~~~~~~~~~~~
+
+Pass a term as a string to the ``Filters.text()`` method to specify the
+term in your text search. 
+
+Example 
+```````
+
+The following example runs a text search on the documents in the
+``fast_and_furious_movies`` collection for titles that contain the
+term "fast".
```

Comment:
Nitpick: Before a code snippet, should the sentence end with ":" or ":"? And is there a difference between the two? If so, what's the difference? Or, should they be consistent as one or the other? 
Applies to all instances.